### DISCUSSION TITLE
v6 to v7 | What is the correct way to pass ref here?

### DISCUSSION THREAD

**Discussion by leochoo** [react-hook-form/react-hook-form#8043] [Q&A]
Since Material UI TextField does not expose ref, I'm supposed to pass it like this
```
const { ref: nameRef, ...nameInputProps } = register("name", {required: true});
<TextField inputRef={nameRef} />
```
...but below works perfectly fine.
```
<TextField {...register("name", {required: true })} />
```

In fact, I had to do it the second way when I use `useFieldArray`, and I had to dynamically generate names.

Why would this work?

p.s. I couldn't find it again but I remember I read Bill Luo saying that the `{...register}` way is not a recommended way anymore when `ref` is not exposed... I was wondering if this still holds true.

**Comment by Moshyfawn** [comment]
The `ref` you need to pass to have RHF manage the input's focus and a couple of other things.

From what I remember, MUI has been forwarding the `ref` to it's root component _(input)_ since some version bump _(sry, can't remember which one)_, so it's fine to use the `register` API with the `TextField` component.

See [this CSB](https://codesandbox.io/s/kind-dawn-ft676y?file=/src/App.tsx)
